lib/nimble_template/helpers/credo.ex

```diff
@@ -1,10 +1,75 @@
 defmodule NimbleTemplate.CredoHelper do
   alias NimbleTemplate.Generator
+  alias NimbleTemplate.Projects.Project
+
+  @do_single_expression_rule_name "CompassCredoPlugin.Check.DoSingleExpression"
+  @single_module_file_rule_name "CompassCredoPlugin.Check.SingleModuleFile"
 
   @spec disable_rule(String.t(), String.t()) :: :ok | {:error, :failed_to_read_file}
   def disable_rule(file_path, rule) do
-    Generator.prepend_content(file_path, """
-    # credo:disable-for-this-file #{rule}
-    """)
+    if File.exists?(file_path) do
+      Generator.prepend_content(file_path, """
+      # credo:disable-for-this-file #{rule}
+      """)
+    end
+  end
+
+  @spec disable_do_single_expression_rule(list()) :: :ok | {:error, :failed_to_read_file}
+  def disable_do_single_expression_rule(file_paths) do
+    Enum.each(file_paths, fn file_path ->
+      disable_rule(file_path, @do_single_expression_rule_name)
+    end)
+  end
+
+  @spec disable_single_module_file_rule(list()) :: :ok | {:error, :failed_to_read_file}
+  def disable_single_module_file_rule(file_paths) do
+    Enum.each(file_paths, fn file_path ->
+      disable_rule(file_path, @single_module_file_rule_name)
+    end)
+  end
+
+  def suppress_credo_warnings_for_base_project(%Project{base_module: base_module}) do
+    disable_do_single_expression_rule(["lib/#{Macro.underscore(base_module)}.ex"])
+  end
```

Comment:
Shall we move this to a `, do:` instead of `do...end` as it only contains one line? This is the convention that the check you're disabling will do in real projects :trollface:

Reply:
Updated in https://github.com/nimblehq/elixir-templates/pull/308/commits/ce27c66a70f9c83889b6fe91e171cd5a03d244e4